fpga_interchange/populate_chip_info.py

```diff
@@ -1740,6 +1760,426 @@ def get_cell(cell_type):
+                        add = False
+                if add:
+                    nets[net[0]] = net[1]
+
+            for cell in old_cell_instances.items():
+                if cell[1].cell_name not in [VCC_cell, GND_cell]:
+                    cell_instances[cell[0]] = cell[1]
+
+            for cell_instance in cell_instances.items():
+                cell_name = cell_instance[1].cell_name
+                instance_to_cell_map[cell_instance[0]] = cell_name
+                graph[cell_instance[0]] = []
+            for net in nets.items():
+                sources = []
+                sinks = []
+                for port in net[1].ports:
+                    if port.instance_name is not None:
+                        cell = instance_to_cell_map[port.instance_name]
+                        if len(cell_pin_direction_map[cell]) == 0:
+                            continue
+                        direction = cell_pin_direction_map[cell][port.name]
+                        if direction:
+                            sources.append(port.instance_name)
+                        else:
+                            sinks.append(port.instance_name)
+                if sources == [] or sinks == []:
+                    continue
+
+                for source in sources:
+                    for sink in sinks:
+                        graph[source].append((sink, 1))
+                        graph[sink].append((source, 0))
+
+            cluster = {}
+            cluster['name'] = macro[0]
+            # If we can't find good root use any cell as root cell
+            key = list(cell_instances.keys())[0]
+            cluster['root_cell_types'] = [cell_instances[key].cell_name]
+            # Check if connection graph is a tree
+
+            if debug:
+                print("\tChecking connections graph")
+
+            subgraphs, cycle = check(graph)
+            assert not cycle, "Code is not design to cope with cycles inside macros"
+            root = None
+            sources = None
+            if subgraphs == 1:
+                # If grah is a tree, search for node, such that when it's remved ther is no connection from any source to any sink
+                root = search_for_root(graph)
+                cluster['root_cell_types'] = [cell_instances[root].cell_name]
+            else:
+                root = source_cells(graph)[0]
+                cluster['root_cell_types'] = [cell_instances[root].cell_name]
+
+            cluster['connection_graph'], cell_idx_map = find_connection_graph(
+                old_nets, cell_instances, cell_pin_direction_map,
+                [VCC_cell, GND_cell], root)
+            if cluster['connection_graph'] is None:
+                continue
+
+            cluster['out_of_site_clusters'], cluster[
+                'phy_graph'] = check_if_in_site(
+                    cell_instances, cell_site_type_map, device,
+                    cell_site_type_to_bels_map, cluster['connection_graph'],
+                    root)
```

Comment:
Looking at `check_if_in_site` it appears that `phy_graph` is always `None`, and it is actually filled later on.

Possibly the function may only return the out_of_site_cluster state and the `phy_graph` may be `None` by default and filled later on.